src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -124,6 +126,22 @@ public bool RemoveActivation(ActivationId act, bool force)
             return Instances.Count == 0;
         }
 
+        public IActivationInfo LookupAndRemoveActivation(ActivationAddress address, bool force)
```

Comment:
this method shares some (non-trivial) logic duplication with the method above (`RemoveActivation`) although they differ slightly in the case of force==true. Is this by design or should they be doing the same and avoiding duplication instead?